modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const NON_NUMERIC_REGEX = /[\D]/g;
+
+export const limitValue = ({
+	defaultValue,
+	max,
+	min,
+	value,
+}: {
+	defaultValue: number;
+	max: number;
+	min: number;
+	value: number;
+}) => {
+	if (isNaN(value) || value < min) {
+		return defaultValue;
+	}
+	else if (value > max) {
+		return max;
+	}
+
+	return value;
+};
+
+export const trimLeftZero: any = ({
```

Comment:
@miltonmc, I put `any` here but I don't know if it's the best for this case and I don't know how to do :worried:

Reply:
Due to the recursion the TypeScript could not infer the return type properly, so we have to enforce the it:
```typescript
export const trimLeftZero: (options: {
	decimalSymbol: string;
	thousandsSeparator: string | void;
	value: string;
}) => string = ({
	decimalSymbol,
	thousandsSeparator,
	value,
}) => {
	...
};
```

Reply:
Thank you!!